wgsl/index.bs

```diff
@@ -261,9 +261,51 @@ that run on the GPU.
 
 <div class='example wgsl global-scope'>
   <xmp highlight='rust'>
+    // A fragment shader which lights textured geometry with point lights.
+
+    // Lights from a storage buffer binding.
+    struct PointLight {
+      position : vec3f,
+      color : vec3f,
+    }
+
+    struct LightStorage {
+      pointCount : u32,
+      point : array<PointLight>,
+    }
+    @group(0) @binding(0) var<storage> lights : LightStorage;
+
+    // Texture and sampler.
+    @group(1) @binding(0) var baseColorSampler : sampler;
+    @group(1) @binding(1) var baseColorTexture : texture_2d<f32>;
+
+    // Function arguments are values from from vertex shader.
     @fragment
-    fn main() -> @location(0) vec4<f32> {
-        return vec4<f32>(0.4, 0.4, 0.8, 1.0);
+    fn fragmentMain(@location(0) worldPos : vec3f,
+                    @location(1) normal : vec3f,
+                    @location(2) uv : vec2f) -> @location(0) vec4f {
+      // Sample the base color of the surface from a texture.
+      let baseColor = textureSample(baseColorTexture, baseColorSampler, uv);
+
+      let N = normalize(normal);
+      var surfaceColor = vec3f(0);
```

Comment:
This *could* be just `vec3f()` because no-arguments means it creates a zero value for the type.
But that's probably less readable in this case.